### PR TITLE
VACMS-4075 Unique validation for content types

### PR DESCRIPTION
**Pull request by ethanteague**
## Description
See #4075

## Testing done
Behat

## QA steps
It's probably overkill, but we could create two of each of the content types to which we've added the `field_enforce_unique_combo` field, and duplicate the values in the unique value combo fields, then verify that a notice is created when trying to save the second one. 
We could probably just spot check a few: E.g.: 
- Create two Facility Health Services with the same field_service_name_and_descripti value, and visually verify that a notice is thrown when saving the second. 
- Create two Story Listing nodes with the same field_office value, and visually verify that a notice is thrown when saving the second. 
- Create two Operating Status nodes with the same field_office value, and visually verify that a notice is thrown when saving the second. 
- Create two Locations Listing nodes with the same field_office value, and visually verify that a notice is thrown when saving the second. 

### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [ ] Tests have been added if necessary.
- [ ] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.

### Select Team for PR review

- [ ] `Core Application Team`
- [x] `Product Support Team`

### Is this PR blocked by another PR?

- [ ] `DO NOT MERGE`

### Does this PR need review from a Product Owner

- [ ] `Needs PO review`

### CMS user-facing annoucement

Is an announcement needed to let editors know of this change? 
- [ ] Yes, and it's written in issue ____ and queued for publication. 
  - [ ] Merge and ping @ rachel-kauff so she's ready to publish after deployment
- [ ] Yes, but it hasn't yet been written 
  - [ ] Don't merge yet -- ping @ rachel-kauff to prompt her to write and queue content
- [ ] No announcement is needed for this code change. 
  - [ ] Merge & carry on unburdened by announcements

### CODE REVIEW

File: tests/behat/drupal-spec-tool/content_model_vamc_content_type_fields.feature [hunk 1]
```diff
@@ -1,197 +1,113 @@
 @api
-Feature: Content model: VAMC Content Type fields
```
Comment (swirtSJW) [inline tests/behat/drupal-spec-tool/content_model_vamc_content_type_fields.feature:2]: @ethanteague  this looks like a mis-past from spectool

File: composer.json [hunk 2]
```diff
@@ -25,6 +25,7 @@
         "drupal/address": "^1.4",
         "drupal/admin_toolbar": "^3.0",
         "drupal/advancedqueue": "^1.0@RC",
+        "drupal/allow_only_one": "1.0.x-dev",
```
Comment (swirtSJW) [inline composer.json:28]: Now that this is released lets get this on the alpha release so it doesn't float.

**Review by swirtSJW — CHANGES_REQUESTED** [review]

File: tests/behat/drupal-spec-tool/content_model_vamc_content_type_fields.feature (hunk 1 above)
Comment (ethanteague) [reply tests/behat/drupal-spec-tool/content_model_vamc_content_type_fields.feature:2]: Dangit. I'm a repeat mis-paste offender. Thanks for catching this, @swirtSJW

**Review by swirtSJW — COMMENTED** [review]
PR looks awesome, but seems behat is still grumpy

File: config/sync/core.entity_form_display.node.health_care_local_health_service.default.yml [hunk 3]
```diff
@@ -149,7 +152,6 @@ third_party_settings:
     group_editorial_workflow:
       children:
         - moderation_state
-        - revision_log
```
Comment (swirtSJW) [inline config/sync/core.entity_form_display.node.health_care_local_health_service.default.yml:152]: gone AWOL.

File: config/sync/core.entity_form_display.node.health_services_listing.default.yml [hunk 4]
```diff
@@ -33,14 +35,13 @@ third_party_settings:
         open: true
         required_fields: true
         weight: 0
-      label: Section settings
+      label: 'Section settings'
       region: content
     group_editorial_workflow:
       children:
         - moderation_state
-        - revision_log
```
Comment (swirtSJW) [inline config/sync/core.entity_form_display.node.health_services_listing.default.yml:41]: AWOL

**Review by swirtSJW — CHANGES_REQUESTED** [review]
Caught a couple `revision_log entries trying to sneak over the fence.

**Review by swirtSJW — APPROVED** [review]
Looks good.   
waiting on tests to finish.

**Comment by swirtSJW** [comment]
@ethanteague It looks like the content build test is probably trying to create a duplicate and gettng blocked  :)
```
@content_editing
  Scenario: Log in and confirm that System-wide alerts can be created and edited # features/content_editing.feature:160
    When I am logged in as a user with the "content_admin" role                  # Drupal\DrupalExtension\Context\DrupalContext::assertAuthenticatedByRole()
    Then I am at "node/add/vamc_operating_status_and_alerts"                     # Drupal\DrupalExtension\Context\MinkContext::assertAtPath()
    And I press "Add new banner alert"                                           # Drupal\DrupalExtension\Context\MinkContext::pressButton()
    And I select "Information" from "Alert type"                                 # Drupal\DrupalExtension\Context\MinkContext::selectOption()
    And I select "Veterans Affairs" from "edit-field-administration"             # Drupal\DrupalExtension\Context\MinkContext::selectOption()
    And I fill in "Title" with "BeHat Alert title"                               # Drupal\DrupalExtension\Context\MinkContext::fillField()
    And I fill in "Alert body" with "BeHat Alert body"                           # Drupal\DrupalExtension\Context\MinkContext::fillField()
    And I press "Save draft and continue editing"                                # Drupal\DrupalExtension\Context\MinkContext::pressButton()
    Then I should see "Pages for the following VAMC systems"                     # Drupal\DrupalExtension\Context\MinkContext::assertPageContainsText()
      The text "Pages for the following VAMC systems" was not found anywhere in the text of the current page. (Behat\Mink\Exception\ResponseTextException)
    │
    And I should not see "BeHat Alert Body"                                      # Drupal\DrupalExtension\Context\MinkContext::assertPageNotContainsText()```

```

**Comment by swirtSJW** [comment]
Actually that ^^ does not seem like it should be affected by this.... but still needs to be checked out.

File: tests/behat/features/content_editing.feature [hunk 5]
```diff
@@ -156,25 +156,20 @@ Feature: CMS Users may effectively create & edit content
     And I should see "City" in the "#edit-field-address-0" element
     And I should see "State" in the "#edit-field-address-0" element
 
-@content_editing
+@content_editing @et
```
Comment (swirtSJW) [inline tests/behat/features/content_editing.feature:159]: ET phone home?

File: tests/behat/features/content_editing.feature [hunk 6]
```diff
@@ -156,25 +156,20 @@ Feature: CMS Users may effectively create & edit content
...
   Scenario: Log in and confirm that System-wide alerts can be created and edited
     When I am logged in as a user with the "content_admin" role
 
     # Create our initial draft
-    Then I am at "node/add/vamc_operating_status_and_alerts"
+    Then I am at "node/1010/edit"
```
Comment (swirtSJW) [inline tests/behat/features/content_editing.feature:164]: This seems like an odd change making it nid specifc.

Comment (ethanteague) [reply tests/behat/features/content_editing.feature:164]: Need to target an existing id - node add forms run the risk of getting shouted down by our Highlander validation, @swirtSJW.

Comment (swirtSJW) [reply tests/behat/features/content_editing.feature:164]: I see why it is needed.   Lets just prefface it with a comment so it is clear what node/1010 is.

**Review by swirtSJW — CHANGES_REQUESTED** [review]
Remove ET and add comment for test   See comments ^^

File: tests/behat/features/content_editing.feature (hunk 6 above)
Comment (ethanteague) [reply tests/behat/features/content_editing.feature:164]: Makes sense to me, @swirtSJW :+1:

**Comment by ethanteague** [comment]
> Remove ET and add comment for test See comments ^^

Updated, @swirtSJW

**Review by swirtSJW — APPROVED** [review]

**Comment by swirtSJW** [comment]
Just triggered a tugboat rebuild to try to get past FE build without failure.

**Comment by swirtSJW** [comment]
That failure is  Notice: MemcachePool::set(): Server memcache (tcp 11211, udp 0) failed with: SERVER_ERROR out of memory storing object
       (130) in /var/lib/tugboat/docroot/modules/contrib/memcache/src/Driver/MemcacheDriver.php line 17

Passes locally.  I am merging.

**Pull request merged**